### PR TITLE
Feature/leave from channel async tests

### PR DESCRIPTION
**Pull request by Gredja**

### CODE REVIEW

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            // Arrange
+            var saasUserId = "A3F16B2A-EEF1-4DAE-8025-D801ED1532A3";
+            var channelId = new Guid("8E976A56-C641-40A6-8B19-5AD726FD5BF7");
+
+            var member = new Member { Id = new Guid("9BA1C9EB-C40F-4C47-9D00-5D797891D34A") };
+            _memberRepositoryMock.Setup(x => x.GetMemberBySaasUserIdAsync(It.Is<string>(saasId => saasId.Equals(saasUserId))))
+                .ReturnsAsync(member)
+                .Verifiable();
+
+            _channelRepositoryMock.Setup(x =>x.IsMemberExistsInChannelAsync(It.Is<Guid>(m => m.Equals(member.Id)), It.Is<Guid>(c =>c.Equals(channelId))))
```
Comment (eugenekirvel) [inline Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: spaces =>

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+            // Assert
+            act.Should().Throw<NetKitChatNotFoundException>()
+                .And.Message.Should().Be($"Unable to leave from channel. Member {nameof(saasUserId)}:{saasUserId} not found.");
+
+            VerifyMocks();
+        }
+
+        [Fact]
+        public void ShouldThrowIfMemberExistsInChannel()
```
Comment (eugenekirvel) [inline Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: wrong name  ShouldThrowIfMemberDoesNotExistInChannel

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            // Assert
+            act.Should().Throw<NetKitChatInvalidOperationException>()
+                .And.Message.Should()
+                .Be($"Unable to leave from channel. Member {nameof(saasUserId)}:{saasUserId} is not joined to channel {nameof(channelId)}:{channelId}.");
+
+            VerifyMocks();
+        }
+
+        [Fact]
+        public async void ShouldReturnTaskCompleted()
```
Comment (eugenekirvel) [inline Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: where is test for success leave from channel

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                    .Verifiable();
+
+                transactionScope.Complete();
+            }
+
+            // Act
+            var result =  _channelService.LeaveFromChannelAsync(saasUserId, channelId); 
+            
+            // Assert
+
```
Comment (eugenekirvel) [inline Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: remove space

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            var member = new Member { Id = new Guid("9BA1C9EB-C40F-4C47-9D00-5D797891D34A") };
+            _memberRepositoryMock.Setup(x => x.GetMemberBySaasUserIdAsync(It.Is<string>(saasId => saasId.Equals(saasUserId))))
+                .ReturnsAsync(member)
+                .Verifiable();
+
+            _channelRepositoryMock.Setup(x => x.IsMemberExistsInChannelAsync(It.Is<Guid>(m => m.Equals(member.Id)), It.Is<Guid>(c => c.Equals(channelId))))
+                .ReturnsAsync(true)
+                .Verifiable();
+
+            using (var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
```
Comment (eugenekirvel) [inline Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: why do we need TransactionScope?

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs (hunk 1 above)
Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: Done

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs (hunk 3 above)
Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: Done

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs (hunk 4 above)
Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: Done

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs (hunk 5 above)
Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: Removed

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs (hunk 2 above)
Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: Done.

Comment (eugenekirvel) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: not done yet

Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/LeaveFromChannelAsyncTests.cs]: Done

**Pull request merged**